src/BinSkim.Rules/EnableSpectreMitigations.cs

```diff
@@ -223,6 +228,8 @@ public override void Analyze(BinaryAnalyzerContext context)
                         {
```

Comment:
Correct me if I'm wrong--this block assumes the minimum compiler version list is always sorted in ascending order, correct?  We should document that here or there.

Reply:
Yes, it makes that assumption, which at present that is correct.  I gave up on a sorted set earlier as I couldn't get it to work and wanted to concentrate on the actual check.  In order to make this robust we will need to either sort the list appropriately such that this assumption is correct, or change the logic to cope with that not being the case.

Reply:
Added a quick TODO to the code such that I can deal with this next iteration

Reply:
Makes sense.  Can we note this in a comment on the structure itself (i.e. "This is assumed to be sorted elsewhere in this check"?), so that any future changes can take that into account?

Reply:
Makes sense - I'll coordinate with @michaelcfanning to make sure this doesn't get missed in the merging of his sanitization with the full experimental rule